ui-modules/utils/logbook/logbook.js

```diff
@@ -39,168 +39,235 @@ export function logbook() {
+            $scope.logEntries = [];
+            doQuery();
+        };
+
+        /**
+         * Converts log entry to string.
+         *
+         * @param {Object} entry The log entry to convert.
+         * @returns {String} log entry converted to string.
+         */
+        vm.covertLogEntryToString = (entry) => {
+            return getCheckedBoxes($scope.logFields).reduce((output, fieldKey) => {
+                    if (entry[fieldKey]) {
+                        output.push(entry[fieldKey])
+                    }
+                    return output;
+                }, []).join(' ');
+        }
+
+        /**
+         * Caches the datetime of the first item in the visible area of the query result.
+         */
+        function cacheDatetimeToScrollTo() {
+            let element = Array.from($element.find('pre')).find(item => item.offsetTop > (autoScrollableElement.scrollTop + autoScrollableElement.offsetTop - 1));
+            let firstLogEntryInTheVisibleArea = $scope.logEntries.find(item => item.id === element.id);
+            if (firstLogEntryInTheVisibleArea) {
+                datetimeToScrollTo = firstLogEntryInTheVisibleArea.datetime;
+            }
+        }
+
+        /**
+         * @returns {boolean} true if current query is a tail request, false otherwise.
+         */
+        function isTail() {
+            return $scope.search.latest && !$scope.search.dateTimeTo;
+        }
 
         /**
          * Performs a logbook query.
          */
         function doQuery() {
-            const levels = $scope.allLevels ? ['ALL'] : getCheckedBoxes($scope.logLevels);
+
+            if (!vm.isValidNumber()) {
+                console.error('number of items is invalid', $scope.search.numberOfItems)
+                return;
+            }
+
+            const levels = getCheckedBoxes($scope.search.logLevels);
 
             const params = {
                 levels: levels,
-                tail: $scope.isLatest,
-                searchPhrase: $scope.searchPhrase,
-                numberOfItems: $scope.numberOfItems,
-                dateTimeFrom: $scope.isLatest ? dateTimeToAutoUpdateFrom : $scope.dateTimeFrom,
-                dateTimeTo: $scope.isLatest ? '' : $scope.dateTimeTo,
+                tail: $scope.search.latest,
+                searchPhrase: $scope.search.phrase,
+                numberOfItems: $scope.search.numberOfItems,
+                dateTimeFrom: isTail() && !isNewQueryParameters ? dateTimeToAutoRefreshFrom : $scope.search.dateTimeFrom,
+                dateTimeTo: $scope.search.dateTimeTo,
             }
 
+            cacheDatetimeToScrollTo();
+
             logbookApi.logbookQuery(params, true).then((logEntries) => {
 
-                if ($scope.isLatest && $scope.logEntries.length !== 0) {
-                    if (logEntries.length > 0) {
+                // Assign unique IDs for new log entries.
+                logEntries.forEach(item => item.id = generateLogEntryId());
```

Comment:
Looks like we need proper IDs assigned in the backend data model to make it reliable, confirmed with @jathanasiou